src/workers/test-explorer/logger.ts

```diff
@@ -1,7 +1,115 @@
+import { OutputChannel, window } from "vscode";
+import { TestResult } from "../../model/test-status.enum";
+
+export const OUTPUT_CHANNEL = "Angular Test Explorer";
+
+export enum LogLevel {
+  ERROR = "error",
+  WARN = "warn",
+  INFO = "info",
+  DEBUG = "debug",
+}
+
+interface Ilog {
+  message: string;
+  date: Date;
+  level: LogLevel;
+}
+
 export class Logger {
-  constructor() {}
+  private static outputChannel: OutputChannel | undefined;
```

Comment:
don't make this static, static classes generate a permanent dependency between classes which makes very difficult unit testing and mocking. If you need this to be only one instance, create it in the AngularTestExplorer and pass it around as a paramenter. That way we can mock it very easily when adding unit tests with Jest.  For the KarmaEventListener for now you can pass it with an extra method if needed. I am also planning of making that a non singleton instance as there is no need for it.